source/common/network/tcp_listener_impl.cc

```diff
@@ -59,7 +59,7 @@ void TcpListenerImpl::onSocketEvent(short flags) {
       break;
     }
 
-    if (rejectCxOverGlobalLimit()) {
+    if (rejectCxOverGlobalLimit() || reject_fraction_.isRandomizedActive(random_)) {
       // The global connection limit has been reached.
       io_handle->close();
       cb_.onReject();
```

Comment:
Looks like this increments  stats_.downstream_global_cx_overflow_.inc()
Think we should have a separate limit for overflow?
If not maybe at least update the docs to note the stat covers both overflow cases.

Reply:
I've added a separate stat.